test/cypress/integration/transactions/transaction-search.cy.js

```diff
@@ -461,4 +462,154 @@ describe('Transactions List', () => {
       cy.get('#download-transactions-link').should('exist')
     })
   })
+
+  describe('Should display relevant error page on search failure ', () => {
+    it('should show error message on a bad request while retrieving the list of transactions', () => {
+      cy.task('setupStubs', [
+        ...sharedStubs(),
+        transactionsStubs.getLedgerTransactionsSuccess({ gatewayAccountId, transactions: unfilteredTransactions })
+      ])
+      cy.visit(transactionsUrl, { failOnStatusCode: false })
+
+      cy.task('clearStubs')
+
+      cy.task('setupStubs', [
+        ...sharedStubs(),
+        transactionsStubs.getLedgerTransactionsFailure(
+          {
+            account_id: gatewayAccountId,
+            limit_total: 'true',
+            limit_total_size: '5001',
+            from_date: '',
+            to_date: '',
+            page: '1',
+            display_size: '100'
+          },
+          400)
+      ])
+
+      // Click the filter button
+      cy.get('#filter').click()
+
+      // Ensure that transaction list is not displayed
+      cy.get('#transactions-list tbody').should('not.exist')
+
+      // Ensure an error message header is displayed
+      cy.get('h1').contains('An error occurred')
+
+      // Ensure a generic error message is displayed
+      cy.get('#errorMsg').contains('Unable to retrieve list of transactions or card types')
+    })
+    it('should display the generic error page, if an internal server error occurs while retrieving the list of transactions', () => {
+      cy.task('setupStubs', [
+        ...sharedStubs(),
+        transactionsStubs.getLedgerTransactionsSuccess({ gatewayAccountId, transactions: unfilteredTransactions })
+      ])
+      cy.visit(transactionsUrl, { failOnStatusCode: false })
+
+      // 1. Filtering FROM
+      // Ensure both the date/time pickers aren't showing
+      cy.get('.datepicker').should('not.exist')
+      cy.get('.ui-timepicker-wrapper').should('not.exist')
+
+      // Fill in a from date
+      cy.get('#fromDate').type('03/5/2018')
+
+      // Fill in a from time
+      cy.get('#fromTime').type('01:00:00')
+
+      // 2. Filtering TO
+
+      // Fill in a to date
+      cy.get('#toDate').type('03/5/2023')
+
+      // Fill in a to time
+      cy.get('#toTime').type('01:00:00')
```

Comment:
I think this can be like the previous test and not have all this stuff.

Reply:
updated as discussed